src/components/AccountDrawer/IconButton.tsx

```diff
@@ -81,4 +92,113 @@ const IconButton = ({ Icon, ...rest }: IconButtonProps | IconLinkProps) => (
   </IconBlock>
 )
 
+type IconWithTextProps = (IconButtonProps | IconLinkProps) & {
+  text: string
+  onConfirm?: () => void
+  onShowConfirm?: (on: boolean) => void
+}
+
+const TextWrapper = styled.div`
+  display: flex;
+  flex-shrink: 0;
+  overflow: hidden;
+  min-width: min-content;
+`
+
+/**
+ * Allows for hiding and showing some text next to an IconButton
+ * Note that for width transitions to animate in CSS we need to always specify the width (no auto)
+ * so there's resize observing and measuring going on here.
+ */
+export const IconWithTextButton = ({ Icon, text, onConfirm, onShowConfirm, onClick, ...rest }: IconWithTextProps) => {
+  const [showText, _setShowText] = useState(false)
+  const frameObserver = useResizeObserver()
+  const hiddenObserver = useResizeObserver()
+
+  const setShowText = useCallback(
+    (val: boolean) => {
+      _setShowText(val)
+      onShowConfirm?.(val)
+    },
+    [onShowConfirm]
+  )
+
+  const dimensionsRef = useRef({
+    frame: 0,
+    hidden: 0,
+  })
+  const dimensions = (() => {
+    // once opened, we avoid updating it to prevent constant resize loop
+    if (!showText) {
+      dimensionsRef.current = { frame: frameObserver.width || 0, hidden: hiddenObserver.width || 0 }
+    }
+    return dimensionsRef.current
+  })()
+
+  // keyboard action to cancel
+  useEffect(() => {
+    if (!showText) return
+    const isClient = typeof window !== 'undefined'
+    if (!isClient) return
+    if (!showText) return
+    const keyHandler = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        setShowText(false)
+        e.preventDefault()
+        e.stopPropagation()
+      }
+    }
+    window.addEventListener('keydown', keyHandler, { capture: true })
+    return () => {
+      window.removeEventListener('keydown', keyHandler, { capture: true })
+    }
+  }, [setShowText, showText])
+
+  const xPad = showText ? 12 : 0
+  const width = showText ? dimensions.frame + dimensions.hidden + xPad : 32
+
+  return (
+    <IconBlock
+      // @ts-ignore its a fine ref
+      ref={frameObserver.ref}
+      {...rest}
+      style={{
+        width,
+        paddingLeft: xPad,
+        paddingRight: xPad,
+      }}
+      onClick={(e: any) => {
```

Comment:
i'm not sure if we have a "no any" lint rule but i would like one 😄

Reply:
for some reason this was not letting me type is properly! if left empty it complained and didn't infer, but adding `MouseEvent` also complained! let me record a video

Reply:
https://github.com/Uniswap/interface/assets/12100/7c11cf57-2f2b-4514-934a-27e013d36041


The crazy part is at the end of all this you still get the original :any error:

<img width="1190" alt="CleanShot 2023-06-02 at 09 36 23@2x" src="https://github.com/Uniswap/interface/assets/12100/fc83ea97-077d-4230-8d28-0d95450e8d0e">

So I'll leave ts-ignore and type it as MouseEvent with a comment.